internal/endpoints/github.go

```diff
@@ -86,16 +78,37 @@ func (eh *EndpointHandler) Github(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	commits, err := convertGithubPayloadToTimelines(eh.conn, payload)
+	service, project, err := getServiceAndProject(eh.conn, payload)
+	if err != nil {
+		if service.ID == 0 { // how do I compare the structs completely?
```

Comment:
Does this make sense? Is there a better way to compare these structs to empty ones?

Reply:
It looks like this code is trying to determine if a service and project were found in the database.  I approach that sort of issue by creating a `NotFoundError` error in the database layer.  In the database layer code, I check to see if a record is found and if its not found then I return an instance of the `NotFoundError`.  An example of this can be found here:  https://github.com/RedHatInsights/cloud-connector/blob/master/internal/connection_repository/sql_connection_locator_v2.go#L100

The `NotFoundError` is located here:  https://github.com/RedHatInsights/cloud-connector/blob/master/internal/connection_repository/types.go#L18

This approach also has a nice effect of hiding the details of the database layer.  In other words, you do not return the `ErrNoRows` error type from the `sql` package into your web/rest api layer.  If needed, you could change the storage mechanism (use redis for example) and keep returning the `NotFoundError` that you define in your db layer.


You might try splitting the `getServiceAndProject()` methods into `getService()` and `getProject()`.  It looks like you are having to explicitly check if both the service and the project are created in this area of the code.  So I don't think you are saving yourself much work by bundling these checks together.  

:thinking:

I hope all of that makes sense.  Ping me if it doesn't and we can hop on a call and walk through it.

Reply:
Wow, I wrote a book there and I'm sorry!  :)